src/main/java/edu/wpi/grip/core/Pipeline.java

```diff
@@ -41,6 +44,21 @@ public void register() {
     }
 
     /**
+     * Remove everything in the pipeline
+     */
+    public void clear() {
+        this.steps.stream()
+                .map(StepRemovedEvent::new)
+                .collect(Collectors.toList())
```

Comment:
You should leave a comment about why you are doing this this way then. Use the above comment.